src/main/java/org/folio/fqm/service/EntityTypeFlatteningService.java

```diff
@@ -109,6 +115,20 @@ private EntityType getFlattenedEntityType(UUID entityTypeId, EntityTypeSource so
     return localizationService.localizeEntityType(flattenedEntityType, sourceFromParent == null);
   }
 
+    private String injectSourceAliasIntoViewExtractor(String sourceViewExtractor, Map<String, String> renamedAliases) {
+    // Convert the keySet to a list and sort it in natural order, then reverse it
+    List<String> aliases = new ArrayList<>(renamedAliases.keySet());
+    aliases.sort(Comparator.naturalOrder());  // Ensure natural order first
```

Comment:
I don't think we should sort the list here. `renamedAliases` is a `LinkedHashMap` so it should already be sorted as a pre-order traversal of the entity type hierarchy, so I wouldn't touch the ordering at all